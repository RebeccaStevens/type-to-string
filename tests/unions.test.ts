import { describe, it } from "vitest";

import { runTests } from "./helpers";

describe("Unions", () => {
  it.each([
    "type Test = string | number",
    "type Test = Readonly<{ foo: string; }> | number;",
    "type Test = readonly string[] | readonly number[];",
    "type Test = { foo: string; } | { bar: string; };",
    "type Test = Readonly<{ foo: string; } | { bar: string; }>;",
  ])("%s", (code) => {
    runTests(code);
  });
});
