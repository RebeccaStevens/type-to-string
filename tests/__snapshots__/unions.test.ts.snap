// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Unions > type Test = { foo: string; } | { bar: string; }; 1`] = `
{
  "typeAlias": "Test",
  "typeReference": null,
  "typeToString": "Test",
}
`;

exports[`Unions > type Test = { foo: string; } | { bar: string; }; 2`] = `
{
  "typeNodeAlias": "Test",
  "typeNodeAsWritten": "{ foo: string; } | { bar: string; }",
  "typeNodeToString": null,
  "typeReferenceNode": null,
}
`;

exports[`Unions > type Test = Readonly<{ foo: string; } | { bar: string; }>; 1`] = `
{
  "typeAlias": "Test",
  "typeReference": null,
  "typeToString": "Test",
}
`;

exports[`Unions > type Test = Readonly<{ foo: string; } | { bar: string; }>; 2`] = `
{
  "typeNodeAlias": "Test",
  "typeNodeAsWritten": "Readonly<{ foo: string; } | { bar: string; }>",
  "typeNodeToString": null,
  "typeReferenceNode": "Readonly<T>",
}
`;

exports[`Unions > type Test = Readonly<{ foo: string; }> | number; 1`] = `
{
  "typeAlias": "Test",
  "typeReference": null,
  "typeToString": "Test",
}
`;

exports[`Unions > type Test = Readonly<{ foo: string; }> | number; 2`] = `
{
  "typeNodeAlias": "Test",
  "typeNodeAsWritten": "Readonly<{ foo: string; }> | number",
  "typeNodeToString": null,
  "typeReferenceNode": null,
}
`;

exports[`Unions > type Test = readonly string[] | readonly number[]; 1`] = `
{
  "typeAlias": "Test",
  "typeReference": null,
  "typeToString": "Test",
}
`;

exports[`Unions > type Test = readonly string[] | readonly number[]; 2`] = `
{
  "typeNodeAlias": "Test",
  "typeNodeAsWritten": "readonly string[] | readonly number[]",
  "typeNodeToString": null,
  "typeReferenceNode": null,
}
`;

exports[`Unions > type Test = string | number 1`] = `
{
  "typeAlias": "Test",
  "typeReference": null,
  "typeToString": "Test",
}
`;

exports[`Unions > type Test = string | number 2`] = `
{
  "typeNodeAlias": "Test",
  "typeNodeAsWritten": "string | number",
  "typeNodeToString": null,
  "typeReferenceNode": null,
}
`;
